### DISCUSSION TITLE
Farmer/Harvester proof checks should timeout

### DISCUSSION THREAD

**Discussion by chrisxch** [Chia-Network/chia-blockchain#11302] [Ideas]
I recently had a failing disk cause very slow lookup times (#11280). That one bad disk rendered tens of thousands of plots unavailable because the farmer/harvester waits patiently for everything to return. I also saw the same scenario occur with unavailable network shares in the early days.

**There is no scenario in which waiting for an already-late proof makes sense.**

**My suggestion is that harvesters should timeout after ~25 seconds and return any proofs/partials found up to that point.**

Example of one plot taking 437 seconds, leading to the entire harvester taking 437 seconds. Any proof found would be invalid and multiple challenges have been skipped entirely:

```
2022-04-23T06:19:30.698 harvester chia.harvester.harvester: WARNING  Looking up qualities on ... took: 437.51397252082825. This should be below 5 seconds to minimize risk of losing rewards.

2022-04-23T06:19:30.698 harvester chia.harvester.harvester: INFO farming a71d9acfe9... Found 0 proofs. Time: 437.51397 s. Total...
```

It would be better to see:
` "WARNING: Looking up qualities on ...plot timed out after 25 seconds."`

**Comment by mariano54** [comment]
Good idea, but we should wait at least 30 (maybe 45) seconds to be conservative

**Reply by NTOHPN** [reply]
How can a PC harvester harvest more than 100 hard disks? The current harvester does not work properly if the hard disk exceeds 50 pieces.

**Reply by mariano54** [reply]
How exactly does it not work? It becomes slower?

**Comment by reb0rn21** [comment]
I must agree I have very slow storage, it works fine when all plots are loaded, but if I update chia and restart harvester I need 24+ for it to load all plots!!
Maybe some option to bypass plot check could be added
